frontend/src/components/elements/PlotlyChart/CustomTheme.tsx

```diff
@@ -426,55 +197,178 @@ export function applyStreamlitThemeTemplateLayout(
+    spec = spec.replaceAll(SEQUENTIAL_6, sequentialColors[6])
+    spec = spec.replaceAll(SEQUENTIAL_7, sequentialColors[7])
+    spec = spec.replaceAll(SEQUENTIAL_8, sequentialColors[8])
+    spec = spec.replaceAll(SEQUENTIAL_9, sequentialColors[9])
+  } else {
+    // Default plotly colors
+    spec = spec.replaceAll(SEQUENTIAL_0, "#0d0887")
+    spec = spec.replaceAll(SEQUENTIAL_1, "#46039f")
+    spec = spec.replaceAll(SEQUENTIAL_2, "#7201a8")
+    spec = spec.replaceAll(SEQUENTIAL_3, "#9c179e")
+    spec = spec.replaceAll(SEQUENTIAL_4, "#bd3786")
+    spec = spec.replaceAll(SEQUENTIAL_5, "#d8576b")
+    spec = spec.replaceAll(SEQUENTIAL_6, "#ed7953")
+    spec = spec.replaceAll(SEQUENTIAL_7, "#fb9f3a")
+    spec = spec.replaceAll(SEQUENTIAL_8, "#fdca26")
+    spec = spec.replaceAll(SEQUENTIAL_9, "#f0f921")
   }
+  return spec
+}
+
+function replaceDivergingColors(
+  spec: string,
+  theme: Theme,
+  elTheme: string
+): string {
+  const DIVERGING_0 = "#000021"
+  const DIVERGING_1 = "#000022"
+  const DIVERGING_2 = "#000023"
+  const DIVERGING_3 = "#000024"
+  const DIVERGING_4 = "#000025"
+  const DIVERGING_5 = "#000026"
+  const DIVERGING_6 = "#000027"
+  const DIVERGING_7 = "#000028"
+  const DIVERGING_8 = "#000029"
+  const DIVERGING_9 = "#000030"
+  const DIVERGING_10 = "#000031"
+
+  if (elTheme === "streamlit") {
+    const divergingColors = getDivergingColorsArray(theme)
+    spec = spec.replaceAll(DIVERGING_0, divergingColors[0])
+    spec = spec.replaceAll(DIVERGING_1, divergingColors[1])
+    spec = spec.replaceAll(DIVERGING_2, divergingColors[2])
+    spec = spec.replaceAll(DIVERGING_3, divergingColors[3])
+    spec = spec.replaceAll(DIVERGING_4, divergingColors[4])
+    spec = spec.replaceAll(DIVERGING_5, divergingColors[5])
+    spec = spec.replaceAll(DIVERGING_6, divergingColors[6])
+    spec = spec.replaceAll(DIVERGING_7, divergingColors[7])
+    spec = spec.replaceAll(DIVERGING_8, divergingColors[8])
+    spec = spec.replaceAll(DIVERGING_9, divergingColors[9])
+    spec = spec.replaceAll(DIVERGING_10, divergingColors[10])
+  } else {
+    // Default plotly colors
+    spec = spec.replaceAll(DIVERGING_0, "#8e0152")
+    spec = spec.replaceAll(DIVERGING_1, "#c51b7d")
+    spec = spec.replaceAll(DIVERGING_2, "#de77ae")
+    spec = spec.replaceAll(DIVERGING_3, "#f1b6da")
+    spec = spec.replaceAll(DIVERGING_4, "#fde0ef")
+    spec = spec.replaceAll(DIVERGING_5, "#f7f7f7")
+    spec = spec.replaceAll(DIVERGING_6, "#e6f5d0")
+    spec = spec.replaceAll(DIVERGING_7, "#b8e186")
+    spec = spec.replaceAll(DIVERGING_8, "#7fbc41")
+    spec = spec.replaceAll(DIVERGING_9, "#4d9221")
+    spec = spec.replaceAll(DIVERGING_10, "#276419")
+  }
+  return spec
+}
+
+export function replaceTemporaryColors(
+  spec: string,
+  theme: Theme,
+  elTheme: string
+): string {
+  spec = spec.replaceAll("#000032", getIncreasingGreen(theme))
+  spec = spec.replaceAll("#000033", getDecreasingRed(theme))
```

Comment:
nit: are all these colors the same between the plotly and streamlit theme, or do we also have to apply different colors for plotly here?

Reply:
maye if this is the case, add a small comment, e.g.:

```typescript
// All the colors here are the same between the streamlit and plotly theme.
```

Reply:
Technically, they are different colors between streamlit and plotly. However, they're still overridable and the color difference is just slightly noticeable. I just thought we can just plainly override and it likely won't be a big deal since we do that with the background colors and other such things in the `layoutWithThemeDefaults` method.